### PR TITLE
[fi] Deterministic parameter generation

### LINKED ISSUE #253: [fi] Deterministic FI parameter sweep
Currently, only a random FI parameter strategy is provided. However, for some gear (e.g. X/Y table for EMFI/LFI) we want to have a deterministic generation strategy.

### PR DESCRIPTION
**Pull request by nasahlpa**
This PR extends the existing FI setup to generate FI parameters deterministically. This is helpful for certain gear where a random parameter generation is not meaningfull (e.g., X/Y table for LFI).

Each FI gear needs to implement the deterministic parameter generation by themselves. This PR extends the Husky VCC glitcher to either randomly or deterministically generate the paramters.

Closes #253.

### CODE REVIEW

File: ci/cfg/ci_ibex_fi_vcc_dummy.yaml [hunk 1]
```diff
@@ -12,6 +12,13 @@ target:
 fisetup:
   fi_gear: "dummy"
   fi_type: "voltage_glitch"
+  # Parameter generation.
+  parameter_generation: "random"
+  # Number of iterations for the parameter sweep.
```
Comment (vogelpi) [inline ci/cfg/ci_ibex_fi_vcc_dummy.yaml:17]: Nit: `# Number of randomized iterations for the parameter sweep.` IIUC, this tells us how many randomized  parameters configs are drawn for each deterministic parameter config. This is nice.

**Review by vogelpi — APPROVED** [review]
This looks very nice @nasahlpa ! I have just one nit but don't mind to change the PR just because of that.

What I like about this implementation is that it looks like being very extensible: If someone decides one more / or one dimension less needs to be deterministic, things should be easily modifiable which is great.

File: ci/cfg/ci_ibex_fi_vcc_dummy.yaml (hunk 1 above)
Comment (nasahlpa) [reply ci/cfg/ci_ibex_fi_vcc_dummy.yaml]: Changed the comment.

File: fault_injection/fi_gear/husky/husky_vcc.py [hunk 2]
```diff
@@ -81,16 +91,32 @@ def generate_fi_parameters(self) -> dict:
...
+            parameters["glitch_width"] = self.curr_glitch_width
+            self.curr_iteration += 1
+        else:
+            raise Exception("HuskyVCC only supports random/deterministic parameter generation")
+
+        # Randomly generate the trigger delay for both cases.
         parameters["trigger_delay"] = random_float_range(self.trigger_delay_min,
                                                          self.trigger_delay_max,
                                                          self.trigger_step)
+
```
[the remark this answers is not in the document]
Comment (nasahlpa) [reply fault_injection/fi_gear/husky/husky_vcc.py]: Removed.

**Comment by nasahlpa** [comment]
Thanks @vogelpi!

**Pull request merged**